pex/scie/science.py

```diff
@@ -0,0 +1,320 @@
+# Copyright 2024 Pex project contributors.
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import
+
+import os.path
+import re
+import shutil
+import subprocess
+from collections import OrderedDict
+from subprocess import CalledProcessError
+
+from pex.atomic_directory import atomic_directory
+from pex.common import is_exe, pluralize, safe_mkdtemp, safe_open
+from pex.compatibility import shlex_quote
+from pex.exceptions import production_assert
+from pex.fetcher import URLFetcher
+from pex.hashing import Sha256
+from pex.layout import Layout
+from pex.pep_440 import Version
+from pex.pex_info import PexInfo
+from pex.scie.model import ScieConfiguration, ScieInfo, SciePlatform, ScieStyle, ScieTarget
+from pex.third_party.packaging.version import InvalidVersion
+from pex.typing import TYPE_CHECKING, cast
+from pex.util import CacheHelper
+from pex.variables import ENV, Variables, unzip_dir_relpath
+
+if TYPE_CHECKING:
+    from typing import Any, Dict, Iterator, Optional, cast
+
+    import attr  # vendor:skip
+    import toml  # vendor:skip
+else:
+    from pex.third_party import attr, toml
+
+
+@attr.s(frozen=True)
+class Manifest(object):
+    target = attr.ib()  # type: ScieTarget
+    path = attr.ib()  # type: str
+
+    def binary_name(self, binary_name):
+        # type: (str) -> str
+        return self.target.platform.binary_name(binary_name)
+
+    def qualified_binary_name(self, binary_name):
+        # type: (str) -> str
+        return self.target.platform.qualified_binary_name(binary_name)
+
+
+MIN_SCIENCE_VERSION = Version("0.3.0")
```

Comment:
Should this be wrapped in a `Version`? That strips trailing 0's, but we use those trailers when trying to pull down a release from GH:

Required:
`https://github.com/a-scie/lift/releases/download/v0.3.0/science-macos-aarch64`

Actual:
```bash
python3 -m pex fastapi uvicorn --scie eager -o foo.pex -- foo.py
https://github.com/a-scie/lift/releases/download/v0.3/science-macos-aarch64
HTTP Error 404: Not Found
```

Reply:
Wow, my face is getting red from all the face-palm. I kept Version, but use its `raw` property when constructing the URL now.